test_osqueryd.py test_daemon_sigint flaky test

```
93: ======================================================================
93: FAIL: test_daemon_sigint (__main__.DaemonTests)
93: ----------------------------------------------------------------------
93: Traceback (most recent call last):
93:   File "test_osqueryd.py", line 153, in test_daemon_sigint
93:     self.daemon_sigint_test_helper(pidfile_path)
93:   File "test_osqueryd.py", line 141, in daemon_sigint_test_helper
93:     self.assertEqual(daemon.retcode, 0)
93: AssertionError: -2 != 0
```

--- a/tools/tests/test_osqueryd.py
+++ b/tools/tests/test_osqueryd.py
@@ -148,8 +148,12 @@ class DaemonTests(test_base.ProcessGenerator, unittest.TestCase):
     def test_daemon_sigint(self):
         pidfile_path = test_base.CONFIG["options"]["pidfile"]
 
-        Path(pidfile_path).touch()
-        self.assertTrue(os.path.exists(pidfile_path))
+        # Ensure there are no leftovers from previous tests
+        try:
+            os.remove(pidfile_path)
+        except:
+            pass
+
         self.daemon_sigint_test_helper(pidfile_path)
 
         try:

